### PR TITLE
Showcase the use of expect_symbol as mentioned in the lecture.

### PR DESCRIPTION
**Pull request by user-0xcafe**
As I mentioned in the lecture today, the use of `expect_symbol`,
instead of the 4 lines that it replaces, can more clearly communicate 
what is happening in the code.

I am curious what you think about it @ckirsch.

### CODE REVIEW

**Comment by ckirsch** [comment]
@user-0xcafe Very nice! Please go ahead and apply that everywhere. I will then pull that in after the current assignment.

**Comment by nbes4** [comment]
Little bit of nitpicking here, but since this addition is meant for more code clarity, would it make sense to rename `expect_symbol` to something like `expect_symbol_and_get()`, `expect_and_get_symbol()` or `expect_symbol_and_consume()` etc? Since, especially for "newcomers", `expect_symbol()` sounds more like a check, but it eventually consumes the current symbol and moves on to the next.

Any opinions on this?
Otherwise, great addition 👍

**Comment by user-0xcafe** [comment]
Great question. I do like the some of your suggestions, especially those that read almost like sentences. 

One reason I didnt use a longer name as that I wanted to mimic `get_symbol`.

The other reason is that all of those names only tell half the story. They do not talk about emitting an error if the symbols do not match. I thought thet this mignt be more confusing. 

But I am happy to change it to `expect_and_get_symbol` if you think that I am overthinking this.

**Comment by nbes4** [comment]
I understand your concerns and the goal to mimic `get_symbol`. In the meantime the name `expect_symbol()` kind of implies that it can fail if we get something that we not "expect", so adding `and_get` would surely be a small bonus for clarity. However, this is just my subjective thinking, so please make sure you name the function as you see fit.

**Comment by ckirsch** [comment]
> I believe that with the second commit I converted all the “low-hanging fruits”. There is a slight variation of the theme that I didn't touch: when the parser exits on a syntax error.

How about another procedure for this case? How often does it occur? How about `force_symbol()` for that?

**Comment by user-0xcafe** [comment]
@ckirsch I just implemented that in the third commit. 

@nbes4 Another possible name that I considered are `get_expected_symbol` and `get_required_symbol`. I would like to get your opinion on those.

**Comment by ckirsch** [comment]
> @ckirsch I just implemented that in the third commit.
> 
> @nbes4 Another possible name that I considered are `get_expected_symbol` and `get_required_symbol`. I would like to get your opinion on those.

@user-0xcafe Nice! Please use those names instead.

**Comment by nbes4** [comment]
`get_expected_symbol `and `get_required_symbol` seem like a good and clear choice!

**Comment by user-0xcafe** [comment]
@ckirsch I changed the names and am very pleased with the way it reads. 
@nbes4 Thanks for your input!

File: selfie.c [hunk 1]
```diff
@@ -417,6 +417,8 @@ uint64_t identifier_string_match(uint64_t string_index);
 uint64_t identifier_or_keyword();
 
 void get_symbol();
+void get_expected_symbol(uint64_t expected_symbol);
+void get_required_symbol(uint64_t expected_symbol);
```
Comment (ckirsch) [inline selfie.c:421]: `expected_symbol` -> `required_symbol`

File: selfie.c [hunk 2]
```diff
@@ -3931,6 +3933,23 @@ void get_symbol() {
...
 }
 
+void get_expected_symbol(uint64_t expected_symbol) {
+  if (symbol == expected_symbol)
+    get_symbol();
+  else
+    syntax_error_symbol(expected_symbol);
+}
+
+void get_required_symbol(uint64_t expected_symbol) {
```
Comment (ckirsch) [inline selfie.c:3943]: call `get_expected_symbol()` here but with a return value that indicates an error on which you then exit.

File: selfie.c [hunk 3]
```diff
@@ -11846,4 +11805,4 @@ int main(int argc, char** argv) {
   exit_code = selfie(0);
 
   return exit_selfie(exit_code, " [ ( -m | -d | -r | -y ) 0-4096 ... ]");
-}
\ No newline at end of file
+}
```
Comment (ckirsch) [inline selfie.c:11808]: no trailing empty line please

**Review by ckirsch — CHANGES_REQUESTED** [review]
@user-0xcafe great work! please fix the remaining issues, I will then pull.

File: selfie.c [hunk 4]
```diff
@@ -3933,21 +3933,23 @@ void get_symbol() {
...
 
-void get_expected_symbol(uint64_t expected_symbol) {
-  if (symbol == expected_symbol)
+uint64_t get_expected_symbol(uint64_t expected_symbol) {
+  uint64_t found;
+
+  if (symbol == expected_symbol) {
     get_symbol();
-  else
+    found = 1;
```
Comment (ckirsch) [inline selfie.c:3941]: Just `return 1` here, no need for `found`.

**Review by ckirsch — CHANGES_REQUESTED** [review]

**Pull request merged**